admin-dev/themes/new-theme/js/components/grid/extension/position-extension.js

```diff
@@ -173,4 +178,67 @@ export default class PositionExtension {
 
     $form.submit();
   }
+
+  /**
+   * Rows have been reordered. This function
+   * finds, for each row ID: the old position, the new position
+   *
+   * @returns {Array}
+   * @private
+   */
+  computeMappingBetweenOldAndNewPositions(rowsData) {
+    const regex = /^row_(\d+)_(\d+)$/;
+
+    const rowsNb = rowsData.length;
+    const positionsBeforeDragAndDrop = {};
+    const positionsAfterDragAndDrop = {};
+    const mapping = [];
+
+    let rowDataMarker;
+    let rowDataParsedData;
+    let i;
+    let rowID;
+    let rowOldPosition;
+    let rowOldOffset;
+    let rowNewOffset;
+
+    // first, compute for each position,
+    // where they were before the drag-and-drop
+    // and where they are after the drag-and-drop
+    for (i = 0; i < rowsNb; i += 1) {
+      rowDataMarker = rowsData[i].rowMarker;
+      rowDataParsedData = regex.exec(rowDataMarker);
+
+      rowID = rowDataParsedData[1];
+      rowOldPosition = parseInt(rowDataParsedData[2], 10);
+      rowOldOffset = rowsData[i].offset;
+      rowNewOffset = i;
+
+      positionsBeforeDragAndDrop[rowOldOffset] = rowOldPosition;
+      positionsAfterDragAndDrop[rowNewOffset] = rowOldPosition;
+    }
+
+    let previousRowPositionWithThisOffset;
+
+    // for each row in table, we look at before the drag-and-drop
+    // and find what other row was there, this is the new position
+    // of current row
+    for (i = 0; i < rowsNb; i += 1) {
+      rowDataMarker = rowsData[i].rowMarker;
+      rowDataParsedData = regex.exec(rowDataMarker);
+      rowID = rowDataParsedData[1];
+      rowOldPosition = parseInt(rowDataParsedData[2], 10);
+
+      rowNewOffset = i;
+      previousRowPositionWithThisOffset = positionsBeforeDragAndDrop[rowNewOffset];
```

Comment:
I'm not sure I get that part, how is the position before drag and drop the new position after 😅